src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncHelpers.CoreCLR.cs

```diff
@@ -274,39 +253,23 @@ void ITaskCompletionAction.Invoke(Task completingTask)
             private struct Ops : IRuntimeAsyncTaskOps<RuntimeAsyncTask<T>>
             {
                 public static Action GetContinuationAction(RuntimeAsyncTask<T> task) => (Action)task.m_action!;
-                public static void MoveNext(RuntimeAsyncTask<T> task) => task.MoveNext();
                 public static Continuation GetContinuationState(RuntimeAsyncTask<T> task) => (Continuation)task.m_stateObject!;
                 public static void SetContinuationState(RuntimeAsyncTask<T> task, Continuation value)
                 {
                     task.m_stateObject = value;
                 }
 
-                public static bool SetCompleted(RuntimeAsyncTask<T> task, Continuation continuation)
+                public static bool SetCompleted(RuntimeAsyncTask<T> task)
                 {
-                    T result;
-                    if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
-                    {
-                        if (typeof(T).IsValueType)
-                        {
-                            result = Unsafe.As<byte, T>(ref continuation.GCData![0]!.GetRawData());
-                        }
-                        else
-                        {
-                            result = Unsafe.As<object, T>(ref continuation.GCData![0]!);
-                        }
-                    }
-                    else
-                    {
-                        result = Unsafe.As<byte, T>(ref continuation.Data![0]);
-                    }
-
-                    return task.TrySetResult(result);
+                    return task.TrySetResult(task.m_result);
```

Comment:
This will assign `m_result = m_result`. That seems ok, but could be optimized a tiny bit in the future, since the result is already there.